### ISSUE TITLE
Missing warning for unawaited coroutine when invoking method in superclass

### ISSUE THREAD

**Issue by mschurr** [python/mypy#14113]
**Bug Report** 

An edge-case where warn-unawaited-coroutine does not work.

**To Reproduce**

```python
class Parent:
    async def func(self) -> bool:
        return True


class Child(Parent):
    async def func(self) -> bool:
        # no warn-unawaited-coroutine emitted here?
        if not super().func():
            return False
        return True
```

**Expected Behavior**
 
Should warn about unawaited coroutine/incompatible type.

**Actual Behavior**

No warning.

**Your Environment**

mypy 0.971

**Comment by erictraut** [comment]
I don't think mypy has a feature to detect unawaited coroutines. In general, it's not a bug to return a coroutine without awaiting it. Using a coroutine in a conditional expression (especially when applying a `not` unary operator) is probably an unintended bug, so maybe that's something mypy could detect. I think that would be a new feature though.

**Comment by mschurr** [comment]
https://mypy.readthedocs.io/en/stable/error_code_list.html#check-that-coroutine-return-value-is-used-unused-coroutine 

I was probably thinking of this.

**Comment by mschurr** [comment]
I've confirmed truthy-function also does not catch this, so maybe this is still a bug?

**Comment by JelleZijlstra** [comment]
The super() calls are a red herring, here's a simpler repro:

```python
async def func() -> bool:
    return True

async def func2() -> None:
    if func():
        print("hi")
```

I think this should be a new feature detecting that `func()` returns a Coroutine object, which is always true. This is similar to but distinct from truthy-function.

**Comment by mschurr-extrahop** [comment]
Worth noting, I guess, that truthy-bool does catch both these cases.

```
"mypy-test.func" returns "Coroutine[Any, Any, bool]" which does not implement __bool__ or __len__ so it could always be true in boolean context  [truthy-bool]
```

**Comment by JelleZijlstra** [comment]
Thanks, I didn't try that code. I think that's enough.

**Comment by ikonst** [comment]
I believe `truthy-bool` is still not default, and might not ever be, which is why we added `truthy-function`. Should we add a `truthy-coroutine`?

**Comment by JukkaL** [comment]
> Should we add a truthy-coroutine?

This sounds reasonable to me. It seems unlikely to produce a false positive, at least if strict optional checking is enabled.

**Comment by ikonst** [comment]
Created #16069.
